lib/ueberauth/strategy/github.ex

```diff
@@ -175,7 +175,7 @@ defmodule Ueberauth.Strategy.Github do
       name: user["name"],
       description: user["bio"],
       nickname: user["login"],
-      email: fetch_email!(user, allow_private_emails),
+      email: maybe_fetch_email(user, allow_private_emails),
```

Comment:
Is there any reason we couldn't update `fetch_email!` to avoid raising an error?

Reply:
This is one of two places where `fetch_email!` used to be used. The other one seems to be a legit place for raising with the expectation of the field existence.

Also, `get_primary_email!` and `get_private_email!` do no raise, so I could clean them up to have no `!` in the names and call `maybe_fetch_email` from `fetch_email!`.